lib/project_types/extension/models/specification_handlers/beacon_extension.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+require "base64"
+require "fileutils"
+require "json"
+require_relative "beacon_extension_utils/script_config"
+require_relative "beacon_extension_utils/script_config_repository"
+
+module Extension
+  module Models
+    module SpecificationHandlers
+      class BeaconExtension < Default
+        SCRIPT_FILE = "build/main.js"
+
+        def name
+          "Beacon Extension"
+        end
+
+        def read_configuration
+        end
+
+        def read_file_contents(context, file_path, &read_file_block)
+          read_file_block.call(file_path)
+        rescue StandardError
+          context.abort(context.message("core.extension.push.beacon_extension.error.file_read_error", file_path))
+        end
+
+        def access_config_property(context, ext_config, key, &process_value)
+          context.abort(context.message("core.extension.push.beacon_extension.error.missing_config_key_error",
+            key)) unless ext_config.key?(key)
+
+          begin
+            process_value.nil? ? ext_config[key] : process_value.call(ext_config[key])
+          rescue StandardError
+            context.abort(context.message("core.extension.push.beacon_extension.error.invalid_config_value_error", key))
+          end
+        end
+
+        def config(context)
+          begin
+            ext_config = BeaconExtensionUtils::ScriptConfigYmlRepository.new(ctx: context).get!.content
+          rescue StandardError
+            context.abort(context.message("core.extension.push.beacon_extension.error.file_read_error",
+              BeaconExtensionUtils::ScriptConfigYmlRepository.filename))
+          end
+
+          script_contents = read_file_contents(context, File.join(context.root, SCRIPT_FILE)) do |file_path|
+            File.read(file_path).chomp
```

Comment:
the `read_file_contents` function does the try catch heavy lifting

Reply:
Can't `File.read` throw an error?

Reply:
that line is happening in the block which is evaluated in the `read_file_contents` function

Reply:
Ah gotcha.. I guess it's sort of misleading because the method is called `read_file_contents` but we are actually doing the read inside the block here..

Would it make sense to include `File.read` as part of `read_file_contents`?